src/v/features/feature_table.cc

```diff
@@ -724,8 +727,19 @@ void feature_table::set_builtin_trial_license(
 
 void feature_table::revoke_license() { _license = std::nullopt; }
 
-const std::optional<security::license>& feature_table::get_license() const {
-    return _license ? _license : _builtin_trial_license;
+const std::optional<security::license>&
+feature_table::get_license(bool use_bootstrap_license) const {
+    if (_license) {
+        return _license;
+    } else if (_builtin_trial_license) {
+        return _builtin_trial_license;
+    } else if (use_bootstrap_license) {
+        static const thread_local std::optional<security::license> ret
+          = _bootstrap_builtin_trial_license;
+        return ret;
+    } else {
+        return _license; // std::nullopt
```

Comment:
nitpick: I'd just return `std::nullopt` explicitly (esp since you've written it out anyway 😛)

Reply:
I chose this because I did not want to declare another static variable since I am returning a const& here. But yeah I could just do that instead.

Reply:
oh, i see it. i didn't parse the return type properly. yeah that's weird. no worries.